frontend/src/Components/SavedBusinesses/SavedBusinesses.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from "react";
+import MainHeader from "../MainHeader/MainHeader";
+import "./SavedBusinesses.css";
+import "../BusinessList/BusinessList.css";
+import BusinessCard from "../BusinessCard/BusinessCard";
+
+const SavedBusinesses = () => {
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const user = JSON.parse(localStorage.getItem("user"));
+  const [businesses, setBusinesses] = useState([]);
+  const [groupedBusinesses, setGroupedBusinesses] = useState({});
+  const apiKey = import.meta.env.VITE_API_KEY;
+
+  const groupBusinessesByCategory = (businesses) => {
+    const grouped = businesses.reduce((acc, business) => {
+      const category = business.category || "Uncategorized";
+      if (!acc[category]) {
+        acc[category] = [];
+      }
+      acc[category].push(business);
+      return acc;
+    }, {});
+    setGroupedBusinesses(grouped);
+  };
+
+  useEffect(() => {
+    const fetchBusinesses = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/saved/${user.id}`
+        );
+        const data = await response.json();
```

Comment:
it seems like there isn't error handling if the response is a 500 status code?  I don't think it will hit the `catch` in this case.

Reply:
Noted